conan/tools/microsoft/msbuilddeps.py

```diff
@@ -152,6 +164,13 @@ def add_valid_ext(libname):
             ext = os.path.splitext(libname)[1]
             return '%s;' % libname if ext in VALID_LIB_EXTENSIONS else '%s.lib;' % libname
 
+        exclude_code_analisys = False
+        if self.exclude_code_analysis:
+            for pattern in self.exclude_code_analysis:
+                if fnmatch.fnmatch(name, pattern):
+                    exclude_code_analisys = True
+                    break
+
```

Comment:
```suggestion
        exclude_code_analysis = False
        if isinstance(self.exclude_code_analysis, list):
              for pattern in self.exclude_code_analysis :
                  if fnmatch.fnmatch(name, pattern):
                      exclude_code_analysis = True
                      break
        else:
             exclude_code_analysis = self.exclude_code_analysis
```

Lets leverage the ``bool`` type, not convert to ``*`` pattern analyze the pattern to retrieve a new ``True``.